src/components/mobile/survey-questions.tsx

```diff
@@ -82,6 +91,83 @@ export function MobileSurveyQuestionnaire({
     },
   });
 
+  useEffect(() => {
+    let PreviouslyOffline = false;
+    const handleOnline = async () => {
+      if (PreviouslyOffline) {
+        try {
+          await SaveResponsesToDatabase(responses, session, submitResponse);
+          toast({
+            title: "Back online!",
+            description:
+              "Your (intermediate) responses have been submitted successfully.",
+          });
+        } catch (error) {
+          toast({
+            title: "Failed to resend responses",
+            description:
+              "An error occurred while attempting to resend responses.",
+            variant: "destructive",
+          });
+        }
+        PreviouslyOffline = false;
+      }
+    };
```

Comment:
I would expect a `useOnlineStatus` effect, a simple boolean state. Then, I would the toaster to be handled higher up on the page. Here I would only expect the answer buffering logic

Reply:
Created #47